cloud/aws/services/ec2/natgateways.go

```diff
@@ -85,12 +89,43 @@ func (s *Service) createNatGateway(subnetID string) (*ec2.NatGateway, error) {
 		AllocationId: aws.String(ip),
 	})
 
-	err = s.EC2.WaitUntilNatGatewayAvailable(&ec2.DescribeNatGatewaysInput{
-		NatGatewayIds: []*string{out.NatGateway.NatGatewayId},
-	})
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to create NAT gateway for subnet ID %q", subnetID)
 	}
 
+	wReq := &ec2.DescribeNatGatewaysInput{NatGatewayIds: []*string{out.NatGateway.NatGatewayId}}
+	if err := s.EC2.WaitUntilNatGatewayAvailable(wReq); err != nil {
+		return nil, errors.Wrapf(err, "failed to wait for nat gateway %q in subnet %q", *out.NatGateway.NatGatewayId, subnetID)
+	}
+
 	return out.NatGateway, nil
 }
+
+func (s *Service) getNatGatewayForSubnet(subnets v1alpha1.Subnets, sn *v1alpha1.Subnet) (string, error) {
+	if sn.IsPublic {
+		return "", errors.Errorf("cannot get NAT gateway for public subnet %q", sn.ID)
```

Comment:
If I'm not mistaken, isn't the NAT gateway actually associated with the public subnet here though? I would expect that we would be interfacing directly with routes only for the private subnets.

Reply:
I should probably improve the error message or the method name, the method is intended to use only with private subnets, the NAT actually lives in a public subnet. Does that make sense?